internal/generator/generator.go

```diff
@@ -11,10 +11,33 @@ import (
 	"io"
 )
 
-// Generate receives a string sourcePath and an io.Writer writer. It will
-// extract all the directives definitions from the .c and .cpp files in
-// sourcePath and its subdirectories, then output the corresponding directive
-// masks map named "directives" and matchFunc named "Match" via writer.
-func Generate(sourcePath string, writer io.Writer, filter map[string]struct{}, override map[string][]Mask) error {
-	return genFromSrcCode(sourcePath, "directives", "Match", writer, filter, override)
+type GenerateConfig struct {
+	Filter map[string]struct{}
+
+	Override map[string][]Mask
+
+	// DirectiveMapName is the name assigned to the variable containing the directives
+	// discovered by generator. The variable name generally starts with a
+	// lowercase to avoid export. Users will use the generated function named by
+	// MatchFuncName to validate the module directives in nginx configurations.
+	DirectiveMapName string
+
+	// MatchFuncName is the name assigned to the matchFunc generated by the generator.
+	// It should generally start with a uppercase to export.
+	// Users will use the generated function named by MatchFuncName
+	// to validate the module directives in nginx configurations.
+	MatchFuncName string
+
+	// MatchFuncComment is the comment appears above the generated MatchFunc.
+	// It may contain some information like what modules are included
+	// in the generated MatchFunc.
```

Comment:
Should it start with `MatchFuncName` to follow Go doc comment conventions, or will you auto generate that part of the comment? What happens if this is left empty?

Reply:
> Should it start with MatchFuncName to follow Go doc comment conventions

Sounds good. I added comments suggesting that it should start with MatchFuncName(but not enforced) in latest commit.

> What happens if this is left empty?

If it is empty, no code comments for `MatchFunc`. I added comments for this in latest commit.